sdks/java/core/src/main/java/org/apache/beam/sdk/coders/RowCoder.java

```diff
@@ -72,6 +75,12 @@ public static RowCoder of(Schema schema) {
   }
```

Comment:
Should this assign a random ID if the provided Schema is already with a UUID?

Reply:
You are correct - this was the cause of the test failures. Fixed now.